std/container/binaryheap.d

```diff
@@ -115,11 +115,26 @@ if (isRandomAccessRange!(Store) || isRandomAccessRange!(typeof(Store.init[])))
     {
         assert(!store.empty, "Cannot pop an empty store.");
         if (store.length == 1) return;
-        auto t1 = moveFront(store[]);
-        auto t2 = moveBack(store[]);
+        static if(isRandomAccessRange!Store)
+        {
+            auto t1 = moveFront(store);
```

Comment:
Yeah, I was thinking something like that.  But then there's functions like pop(Store store).  I guess that I could just leave it be.